### DISCUSSION TITLE
Is there any documentation about SMT-LIB 3?

### DISCUSSION THREAD

**Discussion by srivatsasrinivasmath** [cvc5/cvc5#11092] [General]
Hi,

Is there any documentation about SMT-LIB 3 other than [the proposal](https://smt-lib.org/version3.shtml)? 

Also, has the richer type theory in SMT-LIB 3 lead to any performance gains when compared to SMT-LIB 2?

Best,
Vatsa

**Comment by martin-cs** [comment]
Hi @srivatsasrinivas1729 

Although there is some overlap between "people involved with cvc5" and "people working on SMT-LIB 3", this is not a good place to ask general questions about SMT-LIB.  I would suggest the SMT mailing list.  Also it is worth looking at the presentations at the SMT workshop which update everyone on the progress.

As it is not finished, it is hard to say what gains there have been.  Also, you might need to be a bit more specific about "performance".  Have there been or will there be reductions on solver run time on quantifier-free formula that are already easily described using SMT-LIB 2.0?  I doubt it.  Have there been or will there be reductions in the size of formula needed to encode higher-order and parametrically-typed statements?  Hopefully yes.

HTH

**Reply by srivatsasrinivasmath** [reply]
Thank you for the reply. I will check out the workshops for further information on SMT-LIB 3!
